################################################################################
SHORT DESCRIPTION: 
################################################################################
Add the Mosart 7105 MT multitouch input device (Panasonic CF-C1 systems).

################################################################################
LONG DESCRIPTION: 
################################################################################
Add the Mosart 7105 MT multitouch input device to support Panasonic CF-C1
systems in hid-ids.h list.

################################################################################
CHANGELOG 
################################################################################
Original Author: Ross Philipson <philipsonr@ainfosec.com>
Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>

################################################################################
REMOVAL 
################################################################################
Depending on the HCL, MOsart 7105 MT multitouch input device is supported or
not.

################################################################################
UPSTREAM PLAN
################################################################################
None.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None.

################################################################################
PATCHES 
################################################################################
Index: linux-4.1.6/drivers/hid/hid-ids.h
===================================================================
--- linux-4.1.6.orig/drivers/hid/hid-ids.h	2015-09-11 15:18:57.733926293 +0200
+++ linux-4.1.6/drivers/hid/hid-ids.h	2015-09-11 15:19:34.650091442 +0200
@@ -933,6 +933,7 @@
 #define USB_VENDOR_ID_TURBOX		0x062a
 #define USB_DEVICE_ID_TURBOX_KEYBOARD	0x0201
 #define USB_DEVICE_ID_TURBOX_TOUCHSCREEN_MOSART	0x7100
+#define USB_DEVICE_ID_TURBOX_TOUCHSCREEN_MOSART5 0x7105
 
 #define USB_VENDOR_ID_TWINHAN		0x6253
 #define USB_DEVICE_ID_TWINHAN_IR_REMOTE	0x0100
Index: linux-4.1.6/drivers/hid/hid-multitouch.c
===================================================================
--- linux-4.1.6.orig/drivers/hid/hid-multitouch.c	2015-09-11 15:18:57.733926293 +0200
+++ linux-4.1.6/drivers/hid/hid-multitouch.c	2015-09-11 15:19:34.650091442 +0200
@@ -1269,6 +1269,9 @@
 	{ .driver_data = MT_CLS_CONFIDENCE_MINUS_ONE,
 		MT_USB_DEVICE(USB_VENDOR_ID_TURBOX,
 			USB_DEVICE_ID_TURBOX_TOUCHSCREEN_MOSART) },
+	{ .driver_data = MT_CLS_CONFIDENCE_MINUS_ONE,
+		HID_USB_DEVICE(USB_VENDOR_ID_TURBOX,
+			USB_DEVICE_ID_TURBOX_TOUCHSCREEN_MOSART5) },
 
 	/* Panasonic panels */
 	{ .driver_data = MT_CLS_PANASONIC,
